docs/contracts.rst

```diff
@@ -1211,3 +1212,57 @@ provides methods to interact with contract functions.
 Positional and keyword arguments supplied to the contract caller subclass
 will be used to find the contract function by signature,
 and forwarded to the contract function when applicable.
+
+
+Contract FAQs
+-------------
+
+How do I pass in a struct as a function argument?
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Web3.py accepts struct arguments as dictionaries. This format also supports nested structs.
+Let's take a look at a quick example. Given the following Solidity contract:
+
+.. code-block:: none
+
+   contract Example {
+     address addr;
+
+     struct S1 {
+       address a1;
+       address a2;
+     }
+
+     struct S2 {
+       bytes32 b1;
+       bytes32 b2;
+     }
+
+     struct X {
+       S1 s1;
+       S2 s2;
+       address[] users;
+     }
+
+     function update(X memory x) public {
+       addr = x.s1.a2;
+     }
+
+     function retrieve() public view returns (address) {
+       return addr;
+     }
+   }
+
+You can interact with Web3.py contract API as follows:
+
+.. code-block:: python
+
+   # deploy or lookup the deployed contract, then:
+
+   >>> deployed_contract.functions.retrieve().call()
+   '0x0000000000000000000000000000000000000000'
+
+   >>> deployed_contract.functions.update({'s1': ['0x0000000000000000000000000000000000000001', '0x0000000000000000000000000000000000000002'], 'S2': [b'', b''], 'users': []}).transact()
```

Comment:
Do you have `s1` and `S2` capitalized differently here on purpose? 

And then for the S2 `bytes32` args - in v6 we're planning on making it an error to pass in less than the specified number of bytes, so may be worth passing in 32 bytes here 🤷

Reply:
good catch :)